### PR TITLE
Do not overwrite existing keys when generating new ones.

### LINKED ISSUE #1380: `keys generate`: don't overwrite existing account
### What problem does your feature solve?

Right now if you have an existing account named `alice` and you run `keys generate` again, it will generate a new account, overwrite the old `alice.toml` file with the new seed phrase, and fund the new account.

```
$ cat .soroban/identity/alice.toml
───────┬────────────────────────────────────
       │ File: .soroban/identity/alice.toml
───────┼────────────────────────────────────
   1   │ seed_phrase = "zoo solid..."
───────┴────────────────────────────────────
$ soroban keys generate alice --network local
$ cat .soroban/identity/alice.toml
───────┬────────────────────────────────────
       │ File: .soroban/identity/alice.toml
───────┼────────────────────────────────────
   1   │ seed_phrase = "camp endorse..."
───────┴────────────────────────────────────
```

### What would you like to see?

By default

- check if the account exists and is funded above some level
- If it's not sufficiently funded, fund it (see #1389)
- mention that if you want to regenerate the account with a new seed phrase, you can `keys rm` first

Here's an example of what this could look like:

```
$ soroban keys generate alice --network local
alice already exists!
    seed phrase found at: /path/to/.soroban/identity/alice.toml
    current XLM balance: 231
Balance >100 XLM, nothing to do. 
To generate new keys for alice, you can `keys rm alice` first.
```

Or, if the account doesn't have sufficient balance:

```
$ soroban keys generate alice --network local
alice already exists!
    seed phrase found at: /path/to/.soroban/identity/alice.toml
    account XLM balance: 9
Topping off... Balance is now 1009 XLM.
To generate new keys for alice, you can `keys rm alice` first.
```

### What alternatives are there?

With #1389, we're considering adding a `--minimum-balance` option. I think for the sake of clarity & simplicity, we do NOT add that to `keys generate`. But should we mention the existence of `keys fund` in the output here?

### PR DESCRIPTION
**Pull request by fnando**
### What

`stellar keys generate` will no longer overwrite an identity if it already exists, unless `--overwrite` is specified.

Supersedes #1513. Close #1380.

It differs from #1513 in the sense that it doesn't make any additional checks other than requiring the identity to not exist.

### CODE REVIEW

File: cmd/soroban-cli/src/commands/keys/generate.rs [hunk 1]
```diff
@@ -52,29 +57,47 @@ pub struct Cmd {
...
 impl Cmd {
     pub async fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+
+        if self.config_locator.read_identity(&self.name).is_ok() {
+            if !self.overwrite {
+                return Err(Error::IdentityAlreadyExists(self.name.clone()));
+            }
+
+            print.infoln(format!("Overwriting identity '{}'", &self.name));
```
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/keys/generate.rs:75]: Instead of the info symbol, maybe we need an ❗ to signify destruction. Wdyt?

**Review by leighmcculloch — APPROVED** [review]
One suggestion inline, approved regardless and defer to you on it.

File: cmd/soroban-cli/src/commands/keys/generate.rs (hunk 1 above)
Comment (fnando) [reply cmd/soroban-cli/src/commands/keys/generate.rs:75]: I think it's overkill given that you already provided `--overwrite` explicitly. But I'll do it anyway.

Comment (leighmcculloch) [reply cmd/soroban-cli/src/commands/keys/generate.rs]: That's a good point. And we should probably reserve red icons like ❗ for errors given the red => error signal.

Comment (fnando) [reply cmd/soroban-cli/src/commands/keys/generate.rs]: We do have `error` and `errorln` with ❌ though (so we're good).

**Pull request merged**